cmd/xorgen/internal/tmpl/template.go

```diff
@@ -0,0 +1,198 @@
+package tmpl
+
+import (
+	"bytes"
+	"compress/gzip"
+	_ "embed"
+	"fmt"
+	"github.com/saylorsolutions/gocryptx/pkg/xor"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"text/template"
+	"unicode"
+)
+
+const (
+	idealMinKeyLen = 20
+)
+
+var (
+	//go:embed screen_embed.go.tmpl
+	tmplText     string
+	tmplTemplate = template.Must(template.New("template").Parse(tmplText))
+)
+
+type Params struct {
+	Package        string
+	Exposed        bool
+	Compressed     bool
+	FileMethodName string
+	KeyString      string
+	DataString     string
+	Offset         int
+	keyData        []byte
+	fileData       []byte
+}
+
+func GenerateFile(params *Params, err error) error {
```

Comment:
Yeah, definitely function options. I could see this being confusing later, especially with no documentation on the functions used to populate these parameters.